indico/modules/events/contributions/lists.py

```diff
@@ -80,6 +87,25 @@ def _filter_list_entries(self, query, filters):
             if status_criteria:
                 criteria.append(db.or_(*status_criteria))
 
+        if 'speakers' in filters['items']:
+            contrib_query = (db.session.query(Contribution.id)
+                             .join(ContributionPersonLink)
+                             .join(EventPerson)
+                             .join(User)
+                             .filter(Contribution.event_id == self.event.id)
+                             .filter(User.registrations.any(Registration.is_active &
+                                                            (Registration.event_id == Contribution.event_id))))
```

Comment:
I'd use sqlalchemy's `and_(..)` instead of the `&` operator as soon as it needs to be wrapped on two lines. Wrapping binary operators looks a bit weird ;)

```python
                             .filter(User.registrations.any(and_(Registration.is_active,
                                                                 Registration.event_id == Contribution.event_id))))
```

(`&` is simply a shortcut for it)